[Component templates] Table view

alisonelizabeth opened the pull request:
This PR adds the "Component templates" tab with table view. It also includes the ability to delete a component template.

Create/edit/clone actions and the details flyout will come in subsequent PRs.

**Note to reviewer:** All client code is contained within a new `component_templates` directory under `components`, including associated jest tests.

### How to test
Create >1 component templates and associate one or more with an index template.

For example:

```
PUT _component_template/comp_1
{
  "template": {
    "mappings": {
      "properties": {
        "@timestamp": {
          "type": "date"
        }
      }
    }
  }
}

PUT _component_template/comp_2
{
  "template": {
    "mappings": {
      "properties": {
        "ip_address": {
          "type": "ip"
        }
      }
    }
  }
}

PUT _index_template/template_1
{
  "index_patterns": ["te*", "bar*"],
  "template": {
    "settings": {
      "number_of_shards": 1
    },
    "mappings": {
      "_source": {
        "enabled": false
      },
      "properties": {
        "host_name": {
          "type": "keyword"
        },
        "created_at": {
          "type": "date",
          "format": "EEE MMM dd HH:mm:ss Z yyyy"
        }
      }
    },
    "aliases": {
      "mydata": { }
    }
  },
  "composed_of": ["comp_1", "comp_2"],
}
```

Test cases:
- Empty prompt displays with no component templates
- Delete functionality is disabled if the component template is being used by an index template.
- Delete a single component template.
- Delete more than one component template.
- Filter by templates in use or not in use.
- Reload button refetches component templates data.

### Screenshots
<img width="1256" alt="Screen Shot 2020-06-03 at 11 13 17 AM" src="https://user-images.githubusercontent.com/5226211/83661820-b5ad5580-a594-11ea-97e4-14b4da549dcf.png">
<img width="1279" alt="Screen Shot 2020-06-03 at 11 16 08 AM" src="https://user-images.githubusercontent.com/5226211/83661825-b645ec00-a594-11ea-81e7-6d157d138e6f.png">
<img width="1300" alt="Screen Shot 2020-06-03 at 11 23 43 AM" src="https://user-images.githubusercontent.com/5226211/83661826-b6de8280-a594-11ea-920f-4e6255cde425.png">
<img width="496" alt="Screen Shot 2020-06-03 at 11 15 47 AM" src="https://user-images.githubusercontent.com/5226211/83661822-b645ec00-a594-11ea-90b1-d2bbb173146d.png">
<img width="490" alt="Screen Shot 2020-06-03 at 11 15 56 AM" src="https://user-images.githubusercontent.com/5226211/83661824-b645ec00-a594-11ea-95f6-23943fd1d0bc.png">

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/)
* Commit: 9bd51b88ef4885d8d19a25b43c4fbb39001085b9
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-kibana-ciGroup3](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/execution/node/486/log)
* [Execute xpack-kibana-ciGroup3](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/execution/node/703/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/testReport)

<details><summary>Kibana Pipeline / kibana-xpack-agent / Chrome X-Pack UI Functional Tests.x-pack/test/functional/apps/index_management/home_page·ts.Index Management app Home page Component templates renders the component templates tab</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/testReport//junit/Chrome%20X-Pack%20UI%20Functional%20Tests/x-pack_test_functional_apps_index_management_home_page·ts/Kibana_Pipeline___kibana_xpack_agent___Index_Management_app_Home_page_Component_templates_renders_the_component_templates_tab)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 1 times on tracked branches: https://dryrun

[00:00:00]       │
[00:38:51]         └-: Index Management app
[00:38:51]           └-> "before all" hook
[00:38:51]           └-: Home page
[00:38:51]             └-> "before all" hook
[00:38:51]             └-> "before all" hook
[00:38:51]               │ debg navigating to indexManagement url: http://localhost:6141/app/kibana#/management/data/index_management
[00:38:51]               │ debg navigate to: http://localhost:6141/app/kibana#/management/data/index_management
[00:38:52]               │ debg browser[INFO] http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js 380:106115 "INFO: 2020-06-03T00:38:23Z
[00:38:52]               │        Adding connection to http://localhost:6141/elasticsearch
[00:38:52]               │
[00:38:52]               │      "
[00:38:52]               │ERROR browser[SEVERE] http://localhost:6141/33521/bundles/core/core.entry.js 75:170558 TypeError: Failed to fetch
[00:38:52]               │          at Fetch._callee3$ (http://localhost:6141/33521/bundles/core/core.entry.js:26:69025)
[00:38:52]               │          at l (http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:288:702619)
[00:38:52]               │          at Generator._invoke (http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:288:702372)
[00:38:52]               │          at Generator.forEach.e.<computed> [as throw] (http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:288:702976)
[00:38:52]               │          at fetch_asyncGeneratorStep (http://localhost:6141/33521/bundles/core/core.entry.js:26:63429)
[00:38:52]               │          at _throw (http://localhost:6141/33521/bundles/core/core.entry.js:26:63837)
[00:38:52]               │ debg browser[INFO] http://localhost:6141/login?next=%2Fapp%2Fkibana%3F_t%3D1591144703016#/management/data/index_management 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:38:52]               │
[00:38:52]               │ debg browser[INFO] http://localhost:6141/bundles/app/core/bootstrap.js 11:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:38:52]               │ debg ... sleep(700) start
[00:38:53]               │ debg ... sleep(700) end
[00:38:53]               │ debg returned from get, calling refresh
[00:38:53]               │ debg browser[INFO] http://localhost:6141/login?next=%2Fapp%2Fkibana%3F_t%3D1591144703016#/management/data/index_management 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:38:53]               │
[00:38:53]               │ debg browser[INFO] http://localhost:6141/bundles/app/core/bootstrap.js 11:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:38:53]               │ debg currentUrl = http://localhost:6141/login?next=%2Fapp%2Fkibana%3F_t%3D1591144703016#/management/data/index_management
[00:38:53]               │          appUrl = http://localhost:6141/app/kibana#/management/data/index_management
[00:38:53]               │ debg TestSubjects.find(kibanaChrome)
[00:38:53]               │ debg Find.findByCssSelector('[data-test-subj="kibanaChrome"]') with timeout=60000
[00:38:54]               │ debg browser[INFO] http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js 380:106115 "INFO: 2020-06-03T00:38:25Z
[00:38:54]               │        Adding connection to http://localhost:6141/elasticsearch
[00:38:54]               │
[00:38:54]               │      "
[00:38:54]               │ debg Found login page
[00:38:54]               │ debg TestSubjects.setValue(loginUsername, test_user)
[00:38:54]               │ debg TestSubjects.click(loginUsername)
[00:38:54]               │ debg Find.clickByCssSelector('[data-test-subj="loginUsername"]') with timeout=10000
[00:38:54]               │ debg Find.findByCssSelector('[data-test-subj="loginUsername"]') with timeout=10000
[00:38:55]               │ debg TestSubjects.setValue(loginPassword, changeme)
[00:38:55]               │ debg TestSubjects.click(loginPassword)
[00:38:55]               │ debg Find.clickByCssSelector('[data-test-subj="loginPassword"]') with timeout=10000
[00:38:55]               │ debg Find.findByCssSelector('[data-test-subj="loginPassword"]') with timeout=10000
[00:38:55]               │ debg TestSubjects.click(loginSubmit)
[00:38:55]               │ debg Find.clickByCssSelector('[data-test-subj="loginSubmit"]') with timeout=10000
[00:38:55]               │ debg Find.findByCssSelector('[data-test-subj="loginSubmit"]') with timeout=10000
[00:38:55]               │ debg Find.findByCssSelector('[data-test-subj="kibanaChrome"] nav:not(.ng-hide)') with timeout=60000
[00:38:58]               │ debg browser[INFO] http://localhost:6141/app/kibana?_t=1591144703016#/management/data/index_management 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:38:58]               │
[00:38:58]               │ debg browser[INFO] http://localhost:6141/bundles/app/kibana/bootstrap.js 11:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:38:58]               │ debg browser[INFO] http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js 380:106115 "INFO: 2020-06-03T00:38:28Z
[00:38:58]               │        Adding connection to http://localhost:6141/elasticsearch
[00:38:58]               │
[00:38:58]               │      "
[00:38:58]               │ debg browser[INFO] http://localhost:6141/app/kibana?_t=1591144709255#/management/data/index_management 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:38:58]               │
[00:38:58]               │ debg browser[INFO] http://localhost:6141/bundles/app/kibana/bootstrap.js 11:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:38:58]               │ debg Finished login process currentUrl = http://localhost:6141/app/kibana#/management/data/index_management
[00:38:58]               │ debg TestSubjects.find(kibanaChrome)
[00:38:58]               │ debg Find.findByCssSelector('[data-test-subj="kibanaChrome"]') with timeout=10000
[00:39:00]               │ debg browser[INFO] http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js 380:106115 "INFO: 2020-06-03T00:38:31Z
[00:39:00]               │        Adding connection to http://localhost:6141/elasticsearch
[00:39:00]               │
[00:39:00]               │      "
[00:39:00]               │ debg ... sleep(501) start
[00:39:01]               │ debg ... sleep(501) end
[00:39:01]               │ debg in navigateTo url = http://localhost:6141/app/kibana#/management/data/index_management/indices
[00:39:01]               │ debg TestSubjects.exists(statusPageContainer)
[00:39:01]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="statusPageContainer"]') with timeout=2500
[00:39:03]               │ debg --- retry.tryForTime error: [data-test-subj="statusPageContainer"] is not displayed
[00:39:04]             └-> Loads the app
[00:39:04]               └-> "before each" hook: global before each
[00:39:04]               │ debg Checking for section heading to say Index Management.
[00:39:04]               │ debg TestSubjects.getVisibleText(appTitle)
[00:39:04]               │ debg TestSubjects.find(appTitle)
[00:39:04]               │ debg Find.findByCssSelector('[data-test-subj="appTitle"]') with timeout=10000
[00:39:04]               │ debg TestSubjects.exists(indicesList)
[00:39:04]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="indicesList"]') with timeout=2500
[00:39:04]               │ debg TestSubjects.find(reloadIndicesButton)
[00:39:04]               │ debg Find.findByCssSelector('[data-test-subj="reloadIndicesButton"]') with timeout=10000
[00:39:04]               └- ✓ pass  (93ms) "Index Management app Home page Loads the app"
[00:39:04]             └-: Index templates
[00:39:04]               └-> "before all" hook
[00:39:04]             └-: Component templates
[00:39:04]               └-> "before all" hook
[00:39:04]               └-> renders the component templates tab
[00:39:04]                 └-> "before each" hook: global before each
[00:39:04]                 │ debg TestSubjects.click(component_templatesTab)
[00:39:04]                 │ debg Find.clickByCssSelector('[data-test-subj="component_templatesTab"]') with timeout=10000
[00:39:04]                 │ debg Find.findByCssSelector('[data-test-subj="component_templatesTab"]') with timeout=10000
[00:39:04]                 │ debg isGlobalLoadingIndicatorVisible
[00:39:04]                 │ debg TestSubjects.exists(globalLoadingIndicator)
[00:39:04]                 │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="globalLoadingIndicator"]') with timeout=1500
[00:39:04]                 │ debg TestSubjects.exists(globalLoadingIndicator-hidden)
[00:39:04]                 │ debg Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000
[00:39:05]                 │ debg TestSubjects.exists(componentTemplateList)
[00:39:05]                 │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="componentTemplateList"]') with timeout=2500
[00:39:07]                 │ debg --- retry.tryForTime error: [data-test-subj="componentTemplateList"] is not displayed
[00:39:08]                 │ info Taking screenshot "/dev/shm/workspace/kibana/x-pack/test/functional/screenshots/failure/Index Management app Home page Component templates renders the component templates tab.png"
[00:39:08]                 │ info Current URL is: http://localhost:6141/app/kibana#/management/data/index_management/component_templates
[00:39:08]                 │ info Saving page source to: /dev/shm/workspace/kibana/x-pack/test/functional/failure_debug/html/Index Management app Home page Component templates renders the component templates tab.html
[00:39:08]                 └- ✖ fail: "Index Management app Home page Component templates renders the component templates tab"
[00:39:08]                 │

```

#### Stack Trace
```
Error: expected false to equal true
    at Assertion.assert (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.be.Assertion.equal (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:227:8)
    at Assertion.be (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:69:22)
    at Context.it (test/functional/apps/index_management/home_page.ts:64:42)
```
</details>

---

<details><summary>Kibana Pipeline / kibana-xpack-agent / Chrome X-Pack UI Functional Tests.x-pack/test/functional/apps/index_management/home_page·ts.Index Management app Home page Component templates renders the component templates tab</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/testReport//junit/Chrome%20X-Pack%20UI%20Functional%20Tests/x-pack_test_functional_apps_index_management_home_page·ts/Kibana_Pipeline___kibana_xpack_agent___Index_Management_app_Home_page_Component_templates_renders_the_component_templates_tab_2)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:40:21]         └-: Index Management app
[00:40:21]           └-> "before all" hook
[00:40:21]           └-: Home page
[00:40:21]             └-> "before all" hook
[00:40:21]             └-> "before all" hook
[00:40:21]               │ debg navigating to indexManagement url: http://localhost:6141/app/kibana#/management/data/index_management
[00:40:21]               │ debg navigate to: http://localhost:6141/app/kibana#/management/data/index_management
[00:40:21]               │ debg browser[INFO] http://localhost:6141/login?next=%2Fapp%2Fkibana%3F_t%3D1591142223498#/management/data/index_management 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:40:21]               │
[00:40:21]               │ debg browser[INFO] http://localhost:6141/bundles/app/core/bootstrap.js 11:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:40:21]               │ debg ... sleep(700) start
[00:40:22]               │ debg ... sleep(700) end
[00:40:22]               │ debg returned from get, calling refresh
[00:40:23]               │ debg browser[INFO] http://localhost:6141/login?next=%2Fapp%2Fkibana%3F_t%3D1591142223498#/management/data/index_management 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:40:23]               │
[00:40:23]               │ debg browser[INFO] http://localhost:6141/bundles/app/core/bootstrap.js 11:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:40:23]               │ debg currentUrl = http://localhost:6141/login?next=%2Fapp%2Fkibana%3F_t%3D1591142223498#/management/data/index_management
[00:40:23]               │          appUrl = http://localhost:6141/app/kibana#/management/data/index_management
[00:40:23]               │ debg TestSubjects.find(kibanaChrome)
[00:40:23]               │ debg Find.findByCssSelector('[data-test-subj="kibanaChrome"]') with timeout=60000
[00:40:25]               │ debg browser[INFO] http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js 380:106115 "INFO: 2020-06-02T23:57:06Z
[00:40:25]               │        Adding connection to http://localhost:6141/elasticsearch
[00:40:25]               │
[00:40:25]               │      "
[00:40:25]               │ debg Found login page
[00:40:25]               │ debg TestSubjects.setValue(loginUsername, test_user)
[00:40:25]               │ debg TestSubjects.click(loginUsername)
[00:40:25]               │ debg Find.clickByCssSelector('[data-test-subj="loginUsername"]') with timeout=10000
[00:40:25]               │ debg Find.findByCssSelector('[data-test-subj="loginUsername"]') with timeout=10000
[00:40:25]               │ debg TestSubjects.setValue(loginPassword, changeme)
[00:40:25]               │ debg TestSubjects.click(loginPassword)
[00:40:25]               │ debg Find.clickByCssSelector('[data-test-subj="loginPassword"]') with timeout=10000
[00:40:25]               │ debg Find.findByCssSelector('[data-test-subj="loginPassword"]') with timeout=10000
[00:40:25]               │ debg TestSubjects.click(loginSubmit)
[00:40:25]               │ debg Find.clickByCssSelector('[data-test-subj="loginSubmit"]') with timeout=10000
[00:40:25]               │ debg Find.findByCssSelector('[data-test-subj="loginSubmit"]') with timeout=10000
[00:40:25]               │ debg Find.findByCssSelector('[data-test-subj="kibanaChrome"] nav:not(.ng-hide)') with timeout=60000
[00:40:30]               │ debg browser[INFO] http://localhost:6141/app/kibana?_t=1591142223498#/management/data/index_management 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:40:30]               │
[00:40:30]               │ debg browser[INFO] http://localhost:6141/bundles/app/kibana/bootstrap.js 11:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:40:30]               │ debg browser[INFO] http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js 380:106115 "INFO: 2020-06-02T23:57:11Z
[00:40:30]               │        Adding connection to http://localhost:6141/elasticsearch
[00:40:30]               │
[00:40:30]               │      "
[00:40:30]               │ERROR browser[SEVERE] http://localhost:6141/33521/bundles/plugin/newsfeed/newsfeed.plugin.js 0:18634 TypeError: Failed to fetch
[00:40:30]               │          at Fetch._callee3$ (http://localhost:6141/33521/bundles/commons.bundle.js:3:1156010)
[00:40:30]               │          at l (http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:288:702619)
[00:40:30]               │          at Generator._invoke (http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:288:702372)
[00:40:30]               │          at Generator.forEach.e.<computed> [as throw] (http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:288:702976)
[00:40:30]               │          at asyncGeneratorStep (http://localhost:6141/33521/bundles/commons.bundle.js:3:1149746)
[00:40:30]               │          at _throw (http://localhost:6141/33521/bundles/commons.bundle.js:3:1150143)
[00:40:30]               │ debg browser[INFO] http://localhost:6141/app/kibana?_t=1591142232202#/management/data/index_management 341 Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-P5polb1UreUSOe5V/Pv7tc+yeZuJXiOi/3fqhGsU7BE='), or a nonce ('nonce-...') is required to enable inline execution.
[00:40:30]               │
[00:40:30]               │ debg browser[INFO] http://localhost:6141/bundles/app/kibana/bootstrap.js 11:19 "^ A single error about an inline script not firing due to content security policy is expected!"
[00:40:30]               │ debg Finished login process currentUrl = http://localhost:6141/app/kibana#/management/data/index_management
[00:40:30]               │ debg TestSubjects.find(kibanaChrome)
[00:40:30]               │ debg Find.findByCssSelector('[data-test-subj="kibanaChrome"]') with timeout=10000
[00:40:33]               │ debg browser[INFO] http://localhost:6141/33521/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js 380:106115 "INFO: 2020-06-02T23:57:15Z
[00:40:33]               │        Adding connection to http://localhost:6141/elasticsearch
[00:40:33]               │
[00:40:33]               │      "
[00:40:33]               │ debg ... sleep(501) start
[00:40:34]               │ debg ... sleep(501) end
[00:40:34]               │ debg in navigateTo url = http://localhost:6141/app/kibana#/management/data/index_management/indices
[00:40:34]               │ debg TestSubjects.exists(statusPageContainer)
[00:40:34]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="statusPageContainer"]') with timeout=2500
[00:40:36]               │ debg --- retry.tryForTime error: [data-test-subj="statusPageContainer"] is not displayed
[00:40:37]             └-> Loads the app
[00:40:37]               └-> "before each" hook: global before each
[00:40:37]               │ debg Checking for section heading to say Index Management.
[00:40:37]               │ debg TestSubjects.getVisibleText(appTitle)
[00:40:37]               │ debg TestSubjects.find(appTitle)
[00:40:37]               │ debg Find.findByCssSelector('[data-test-subj="appTitle"]') with timeout=10000
[00:40:37]               │ debg TestSubjects.exists(indicesList)
[00:40:37]               │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="indicesList"]') with timeout=2500
[00:40:37]               │ debg TestSubjects.find(reloadIndicesButton)
[00:40:37]               │ debg Find.findByCssSelector('[data-test-subj="reloadIndicesButton"]') with timeout=10000
[00:40:37]               └- ✓ pass  (97ms) "Index Management app Home page Loads the app"
[00:40:37]             └-: Index templates
[00:40:37]               └-> "before all" hook
[00:40:37]             └-: Component templates
[00:40:37]               └-> "before all" hook
[00:40:37]               └-> renders the component templates tab
[00:40:37]                 └-> "before each" hook: global before each
[00:40:37]                 │ debg TestSubjects.click(component_templatesTab)
[00:40:37]                 │ debg Find.clickByCssSelector('[data-test-subj="component_templatesTab"]') with timeout=10000
[00:40:37]                 │ debg Find.findByCssSelector('[data-test-subj="component_templatesTab"]') with timeout=10000
[00:40:37]                 │ debg isGlobalLoadingIndicatorVisible
[00:40:37]                 │ debg TestSubjects.exists(globalLoadingIndicator)
[00:40:37]                 │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="globalLoadingIndicator"]') with timeout=1500
[00:40:37]                 │ debg TestSubjects.exists(globalLoadingIndicator-hidden)
[00:40:37]                 │ debg Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000
[00:40:38]                 │ debg TestSubjects.exists(componentTemplateList)
[00:40:38]                 │ debg Find.existsByDisplayedByCssSelector('[data-test-subj="componentTemplateList"]') with timeout=2500
[00:40:40]                 │ debg --- retry.tryForTime error: [data-test-subj="componentTemplateList"] is not displayed
[00:40:41]                 │ info Taking screenshot "/dev/shm/workspace/kibana/x-pack/test/functional/screenshots/failure/Index Management app Home page Component templates renders the component templates tab.png"
[00:40:41]                 │ info Current URL is: http://localhost:6141/app/kibana#/management/data/index_management/component_templates
[00:40:41]                 │ info Saving page source to: /dev/shm/workspace/kibana/x-pack/test/functional/failure_debug/html/Index Management app Home page Component templates renders the component templates tab.html
[00:40:41]                 └- ✖ fail: "Index Management app Home page Component templates renders the component templates tab"
[00:40:41]                 │

```

#### Stack Trace
```
Error: expected false to equal true
    at Assertion.assert (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.be.Assertion.equal (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:227:8)
    at Assertion.be (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:69:22)
    at Context.it (test/functional/apps/index_management/home_page.ts:64:42)
```
</details>

---

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/","number":"51250","commit":"9bd51b88ef4885d8d19a25b43c4fbb39001085b9"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

alisonelizabeth commented on x-pack/plugins/index_management/public/application/components/component_templates/__jest__/client_integration/component_template_list.test.ts:
```diff
@@ -0,0 +1,177 @@
+      testBed = await setup();
+
+      await act(async () => {
+        await nextTick(100);
```
@sebelga what's the best way to avoid using `nextTick()` in these tests, but still account for latency with the mocked request?

sebelga replied:
@alisonelizabeth Look at index_template_tab.test.ts in master on L37 (https://github.com/elastic/kibana/commit/c77e9bf8f88eded36028a86d49f77d1f5ebaf061#diff-29885e01c1393c73c1e16cf4a07fad91R37) I've removed all `nextTick()` calls.

You probably need to wrap the mounting (`setup()` call) inside the `act()` call.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/)
* Commit: ee7293d40a60254ea91205161762a558422a8361
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/execution/node/534/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/execution/node/689/log)
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/execution/node/289/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/testReport)

<details><summary>Kibana Pipeline / kibana-xpack-agent / X-Pack API Integration Tests.x-pack/test/api_integration/apis/management/index_management/component_templates·ts.apis management index management Component templates Get all component templates should return an array of component templates</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_management_index_management_component_templates·ts/Kibana_Pipeline___kibana_xpack_agent___apis_management_index_management_Component_templates_Get_all_component_templates_should_return_an_array_of_component_templates)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 1 times on tracked branches: https://dryrun

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:05:02]           └-: management
[00:05:02]             └-> "before all" hook
[00:05:25]             └-: index management
[00:05:25]               └-> "before all" hook
[00:05:29]               └-: Component templates
[00:05:29]                 └-> "before all" hook
[00:05:29]                 └-: Get
[00:05:29]                   └-> "before all" hook
[00:05:29]                   └-> "before all" hook
[00:05:29]                     │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591203739897873236] adding component template [test_component_template]
[00:05:29]                   └-: all component templates
[00:05:29]                     └-> "before all" hook
[00:05:29]                     └-> should return an array of component templates
[00:05:29]                       └-> "before each" hook: global before each
[00:05:29]                       └- ✖ fail: "apis management index management Component templates Get all component templates should return an array of component templates"
[00:05:29]                       │

```

#### Stack Trace
```
{ Error: expected { name: 'test_component_template',
  usedBy: [],
  hasSettings: true,
  hasMappings: true,
  hasAliases: false } to sort of equal { name: 'test_component_template',
  isInUse: false,
  hasSettings: true,
  hasMappings: true,
  hasAliases: false }
    at Assertion.assert (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.eql (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:244:8)
    at Context.it (test/api_integration/apis/management/index_management/component_templates.ts:62:44)
  actual:
   '{\n  "hasAliases": false\n  "hasMappings": true\n  "hasSettings": true\n  "name": "test_component_template"\n  "usedBy": []\n}',
  expected:
   '{\n  "hasAliases": false\n  "hasMappings": true\n  "hasSettings": true\n  "isInUse": false\n  "name": "test_component_template"\n}',
  showDiff: true }
```
</details>

---

<details><summary>Kibana Pipeline / kibana-xpack-agent / X-Pack API Integration Tests.x-pack/test/api_integration/apis/management/index_management/component_templates·ts.apis management index management Component templates Get all component templates should return an array of component templates</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_management_index_management_component_templates·ts/Kibana_Pipeline___kibana_xpack_agent___apis_management_index_management_Component_templates_Get_all_component_templates_should_return_an_array_of_component_templates_2)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:05:15]           └-: management
[00:05:15]             └-> "before all" hook
[00:05:39]             └-: index management
[00:05:39]               └-> "before all" hook
[00:05:44]               └-: Component templates
[00:05:44]                 └-> "before all" hook
[00:05:44]                 └-: Get
[00:05:44]                   └-> "before all" hook
[00:05:44]                   └-> "before all" hook
[00:05:44]                     │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-ubuntu-16-tests-xl-1591203739897873236] adding component template [test_component_template]
[00:05:44]                   └-: all component templates
[00:05:44]                     └-> "before all" hook
[00:05:44]                     └-> should return an array of component templates
[00:05:44]                       └-> "before each" hook: global before each
[00:05:44]                       └- ✖ fail: "apis management index management Component templates Get all component templates should return an array of component templates"
[00:05:44]                       │

```

#### Stack Trace
```
{ Error: expected { name: 'test_component_template',
  usedBy: [],
  hasSettings: true,
  hasMappings: true,
  hasAliases: false } to sort of equal { name: 'test_component_template',
  isInUse: false,
  hasSettings: true,
  hasMappings: true,
  hasAliases: false }
    at Assertion.assert (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.eql (/dev/shm/workspace/kibana/packages/kbn-expect/expect.js:244:8)
    at Context.it (test/api_integration/apis/management/index_management/component_templates.ts:62:44)
  actual:
   '{\n  "hasAliases": false\n  "hasMappings": true\n  "hasSettings": true\n  "name": "test_component_template"\n  "usedBy": []\n}',
  expected:
   '{\n  "hasAliases": false\n  "hasMappings": true\n  "hasSettings": true\n  "isInUse": false\n  "name": "test_component_template"\n}',
  showDiff: true }
```
</details>

---

### History
* :broken_heart: [Build #51250](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/) failed 9bd51b88ef4885d8d19a25b43c4fbb39001085b9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/","number":"51250","commit":"9bd51b88ef4885d8d19a25b43c4fbb39001085b9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/","number":"51480","commit":"ee7293d40a60254ea91205161762a558422a8361"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/)
* Commit: f8dd123d956540d4536a15b7b59de2e847cd1bd1
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/execution/node/293/log)

### History
* :broken_heart: [Build #51480](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/) failed ee7293d40a60254ea91205161762a558422a8361
* :broken_heart: [Build #51250](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/) failed 9bd51b88ef4885d8d19a25b43c4fbb39001085b9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/","number":"51250","commit":"9bd51b88ef4885d8d19a25b43c4fbb39001085b9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/","number":"51480","commit":"ee7293d40a60254ea91205161762a558422a8361"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/","number":"51544","commit":"f8dd123d956540d4536a15b7b59de2e847cd1bd1"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/)
* Commit: 63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0

### History
* :broken_heart: [Build #51544](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/) failed f8dd123d956540d4536a15b7b59de2e847cd1bd1
* :broken_heart: [Build #51480](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/) failed ee7293d40a60254ea91205161762a558422a8361
* :broken_heart: [Build #51250](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/) failed 9bd51b88ef4885d8d19a25b43c4fbb39001085b9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/","number":"51250","commit":"9bd51b88ef4885d8d19a25b43c4fbb39001085b9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/","number":"51480","commit":"ee7293d40a60254ea91205161762a558422a8361"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/","number":"51544","commit":"f8dd123d956540d4536a15b7b59de2e847cd1bd1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/","number":"51635","commit":"63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/)
* Commit: 07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/execution/node/289/log)

### History
* :green_heart: [Build #51635](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/) succeeded 63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0
* :broken_heart: [Build #51544](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/) failed f8dd123d956540d4536a15b7b59de2e847cd1bd1
* :broken_heart: [Build #51480](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/) failed ee7293d40a60254ea91205161762a558422a8361
* :broken_heart: [Build #51250](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/) failed 9bd51b88ef4885d8d19a25b43c4fbb39001085b9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/","number":"51250","commit":"9bd51b88ef4885d8d19a25b43c4fbb39001085b9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/","number":"51480","commit":"ee7293d40a60254ea91205161762a558422a8361"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/","number":"51544","commit":"f8dd123d956540d4536a15b7b59de2e847cd1bd1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/","number":"51635","commit":"63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/","number":"52156","commit":"07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52213/)
* Commit: d0fc261579a5aba87c44281d2ec640509c7472d2

### History
* :broken_heart: [Build #52156](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/) failed 07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a
* :green_heart: [Build #51635](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/) succeeded 63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0
* :broken_heart: [Build #51544](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/) failed f8dd123d956540d4536a15b7b59de2e847cd1bd1
* :broken_heart: [Build #51480](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/) failed ee7293d40a60254ea91205161762a558422a8361
* :broken_heart: [Build #51250](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/) failed 9bd51b88ef4885d8d19a25b43c4fbb39001085b9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51250/","number":"51250","commit":"9bd51b88ef4885d8d19a25b43c4fbb39001085b9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/","number":"51480","commit":"ee7293d40a60254ea91205161762a558422a8361"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/","number":"51544","commit":"f8dd123d956540d4536a15b7b59de2e847cd1bd1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/","number":"51635","commit":"63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/","number":"52156","commit":"07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52213/","number":"52213","commit":"d0fc261579a5aba87c44281d2ec640509c7472d2"}]}
PIPELINE-->

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52494/)
* Commit: 87ca2e93a26e5b1af10b5dbcf37e00d8326e477c

### History
* :green_heart: [Build #52213](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52213/) succeeded d0fc261579a5aba87c44281d2ec640509c7472d2
* :broken_heart: [Build #52156](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/) failed 07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a
* :green_heart: [Build #51635](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/) succeeded 63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0
* :broken_heart: [Build #51544](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/) failed f8dd123d956540d4536a15b7b59de2e847cd1bd1
* :broken_heart: [Build #51480](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/) failed ee7293d40a60254ea91205161762a558422a8361

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51480/","number":"51480","commit":"ee7293d40a60254ea91205161762a558422a8361"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/","number":"51544","commit":"f8dd123d956540d4536a15b7b59de2e847cd1bd1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/","number":"51635","commit":"63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/","number":"52156","commit":"07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52213/","number":"52213","commit":"d0fc261579a5aba87c44281d2ec640509c7472d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52494/","number":"52494","commit":"87ca2e93a26e5b1af10b5dbcf37e00d8326e477c"}]}
PIPELINE-->

sebelga commented:
Great job @alisonelizabeth ! I will review the code now but leaving here a comment about UX.

I am not sure about repeating in each cell "index template" under the "Used by" column. I wonder if by just having the number (e.g. `3`) it would be enough. In a following iteration, we could add a link on that number that would point to the index template table filtered on that component. This is similar behaviour that @cjcenizal demoed yesterday with data streams. WDYT?

Also, the cross icon looks like there is an error. What do you think about not having the icon, and setting the text in italic with lighter text color like this?

<img width="718" alt="Screenshot 2020-06-09 at 11 27 23" src="https://user-images.githubusercontent.com/2854616/84130959-37452d80-aa44-11ea-934d-5069f2588a24.png">

In a following iteration, as an enhancement, I can see that the current "in use/not in use" filter is replaced by a dropdown select with those 2 options + 3 others

- With mapping
- With settings
- With aliases

sebelga commented on x-pack/plugins/index_management/public/application/components/component_templates/component_template_list/delete_modal.tsx:
```diff
@@ -0,0 +1,128 @@
+      .then(({ data: { itemsDeleted, errors }, error }) => {
+        const hasDeletedComponentTemplates = itemsDeleted && itemsDeleted.length;
+
+        if (hasDeletedComponentTemplates) {
```
nit: this is the typical example where I prefer to have an `i18nTexts` object declared on top of the component and use it in the JSX. It makes things much clearer to understand IMO

alisonelizabeth replied:
👍 I'll keep this in mind. I'm going to leave as is for now, since I am not following this pattern elsewhere.

sebelga commented on x-pack/plugins/index_management/public/application/components/component_templates/component_template_list/table.tsx:
```diff
@@ -0,0 +1,224 @@
+      onSelectionChange: setSelection,
+      selectable: ({ usedBy }) => usedBy.length === 0,
+      selectableMessage: (selectable) =>
+        !selectable
```
nit: I prefer to use the affirmative in ternary

alisonelizabeth replied:
Agreed

sebelga commented on x-pack/plugins/index_management/public/application/components/component_templates/component_template_list/table.tsx:
```diff
@@ -0,0 +1,224 @@
+      const { field } = column as EuiTableFieldDataColumnType<ComponentTemplateListItem>;
+
+      return {
+        'data-test-subj': `componentTemplateTableRow-${field}`,
```
Is all the logic in `cellProps ` necessary?

alisonelizabeth replied:
This allows the name of the cell to be part of the `data-test-subj`. I see I'm not currently using this in any of the tests though, so I removed for now.

sebelga commented on x-pack/plugins/index_management/public/application/components/component_templates/lib/documentation.ts:
```diff
@@ -0,0 +1,20 @@
+  const docsBase = `${ELASTIC_WEBSITE_URL}guide/en`;
+  const esDocsBase = `${docsBase}/elasticsearch/reference/${DOC_LINK_VERSION}`;
+
+  function getComponentTemplatesLink() {
```
Why not use an object instead of declaring methods? Do we expect parameters to be passed?

```js
export const getDocumentation = ({ ELASTIC_WEBSITE_URL, DOC_LINK_VERSION }: DocLinksSetup) => {
  const docsBase = `${ELASTIC_WEBSITE_URL}guide/en`;
  const esDocsBase = `${docsBase}/elasticsearch/reference/${DOC_LINK_VERSION}`;

  return {
    componentTemplates: `${esDocsBase}/indices-component-template.html`,
  };
};

```

alisonelizabeth replied:
Great point. Fixed.

sebelga commented on x-pack/plugins/index_management/public/application/index.tsx:
```diff
@@ -22,15 +25,26 @@ export const renderApp = (
+    httpClient: services.httpService.httpClient,
+    apiBasePath: API_BASE_PATH,
+    appBasePath: BASE_PATH,
+    trackMetric: services.uiMetricService.trackMetric.bind(services.uiMetricService),
```
Nice! 👍

sebelga commented on x-pack/plugins/index_management/server/client/elasticsearch.ts:
```diff
@@ -60,4 +61,14 @@ export const elasticsearchJsPlugin = (Client: any, config: any, components: any)
     method: 'DELETE',
   });
+
+  // Index templates v2
```
Nit: Can we update the comment `// Composable templates`?

alisonelizabeth commented on x-pack/plugins/index_management/common/lib/component_template_serialization.ts:
```diff
@@ -0,0 +1,63 @@
+
+const hasEntries = (data: object = {}) => Object.entries(data).length > 0;
+
+const getAssociatedIndexTemplates = (
```
Great point. I went ahead and addressed this.

alisonelizabeth commented on x-pack/plugins/index_management/common/types/templates.ts:
```diff
@@ -49,6 +49,11 @@ export interface TemplateDeserialized {
   };
 }
 
+export interface TemplateV2Es {
```
Good catch. I forgot to update this when pulling your changes.

alisonelizabeth commented:
Thanks for the review @sebelga! I believe I addressed all of your feedback.

> I am not sure about repeating in each cell "index template" under the "Used by" column. I wonder if by just having the number (e.g. 3) it would be enough. In a following iteration, we could add a link on that number that would point to the index template table filtered on that component. This is similar behaviour that @cjcenizal demoed yesterday with data streams. WDYT?

> Also, the cross icon looks like there is an error. What do you think about not having the icon, and setting the text in italic with lighter text color like this?

Thanks for bringing this up. I was thinking the same thing after seeing CJ's changes. I went ahead and followed your suggestions. I changed the column title from "Used by" to "Index templates" to make it a little more clear what the number represents. There might still be some better copy we can use here; I'll follow up on this during copy review.

<img width="1265" alt="Screen Shot 2020-06-09 at 10 47 14 AM" src="https://user-images.githubusercontent.com/5226211/84163634-b5530580-aa3f-11ea-8ede-2d3827ae37d5.png">

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52729/)
* Commit: 5ddb122d33b6be8dc0fc0512c3468746c76908c0
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52729/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute x-pack-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52729/execution/node/289/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52729/testReport)

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/plugins/index_management/public/application/components/component_templates/__jest__/client_integration.<ComponentTemplateList /> With component templates should render the list view</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52729/testReport//junit/X-Pack%20Jest%20Tests/x-pack_plugins_index_management_public_application_components_component_templates___jest___client_integration/Kibana_Pipeline___x_pack_intake_agent____ComponentTemplateList____With_component_templates_should_render_the_list_view)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
Error: expect(received).toEqual(expected) // deep equality

- Expected
+ Received

  Array [
    "",
    "test_component_template_2",
-   "1 index template",
+   "1",
    "",
    "",
    "",
    "",
  ]
    at forEach (/dev/shm/workspace/kibana/x-pack/plugins/index_management/public/application/components/component_templates/__jest__/client_integration/component_template_list.test.ts:67:21)
    at Array.forEach (<anonymous>)
    at Object.test (/dev/shm/workspace/kibana/x-pack/plugins/index_management/public/application/components/component_templates/__jest__/client_integration/component_template_list.test.ts:63:24)
    at Object.asyncJestTest (/dev/shm/workspace/kibana/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/dev/shm/workspace/kibana/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/dev/shm/workspace/kibana/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/dev/shm/workspace/kibana/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
```
</details>

---

### History
* :green_heart: [Build #52494](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52494/) succeeded 87ca2e93a26e5b1af10b5dbcf37e00d8326e477c
* :green_heart: [Build #52213](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52213/) succeeded d0fc261579a5aba87c44281d2ec640509c7472d2
* :broken_heart: [Build #52156](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/) failed 07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a
* :green_heart: [Build #51635](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/) succeeded 63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0
* :broken_heart: [Build #51544](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/) failed f8dd123d956540d4536a15b7b59de2e847cd1bd1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51544/","number":"51544","commit":"f8dd123d956540d4536a15b7b59de2e847cd1bd1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/","number":"51635","commit":"63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/","number":"52156","commit":"07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52213/","number":"52213","commit":"d0fc261579a5aba87c44281d2ec640509c7472d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52494/","number":"52494","commit":"87ca2e93a26e5b1af10b5dbcf37e00d8326e477c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52729/","number":"52729","commit":"5ddb122d33b6be8dc0fc0512c3468746c76908c0"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52759/)
* Commit: 8d2222ec81ed03cad23bec36b320ff5d541f7c60

### History
* :broken_heart: [Build #52729](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52729/) failed 5ddb122d33b6be8dc0fc0512c3468746c76908c0
* :green_heart: [Build #52494](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52494/) succeeded 87ca2e93a26e5b1af10b5dbcf37e00d8326e477c
* :green_heart: [Build #52213](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52213/) succeeded d0fc261579a5aba87c44281d2ec640509c7472d2
* :broken_heart: [Build #52156](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/) failed 07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a
* :green_heart: [Build #51635](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/) succeeded 63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51635/","number":"51635","commit":"63789563b99f05a26dc0c4f6fdf26f7a3ffb19b0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52156/","number":"52156","commit":"07aeb9bbaf4df2ab53454b9e267e5a3bfa3c4a8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52213/","number":"52213","commit":"d0fc261579a5aba87c44281d2ec640509c7472d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52494/","number":"52494","commit":"87ca2e93a26e5b1af10b5dbcf37e00d8326e477c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52729/","number":"52729","commit":"5ddb122d33b6be8dc0fc0512c3468746c76908c0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52759/","number":"52759","commit":"8d2222ec81ed03cad23bec36b320ff5d541f7c60"}]}
PIPELINE-->
